mithril-common/src/chain_observer/pallas_observer.rs

```diff
@@ -239,6 +231,41 @@ impl PallasChainObserver {
         Ok(Some(stake_distribution))
     }
 
+    /// Fetches chain point and genesis config through the local statequery.
+    /// The KES period is calculated afterwards.
+    async fn calculate_kes_period(
+        &self,
+        client: &mut NodeClient,
+    ) -> Result<Option<KESPeriod>, ChainObserverError> {
+        let statequery = client.statequery();
+
+        statequery
+            .acquire(None)
+            .await
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to acquire statequery")?;
+
+        let chain_point = queries_v16::get_chain_point(statequery)
+            .await
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to get chain point")?;
+
+        let era = queries_v16::get_current_era(statequery)
+            .await
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to get current era")?;
+
+        let genesis_config = queries_v16::get_genesis_config(statequery, era)
+            .await
+            .map_err(|err| anyhow!(err))
+            .with_context(|| "PallasChainObserver failed to get genesis config")?;
+
+        let slots_per_kes_period = genesis_config[0].slots_per_kes_period as u64;
```

Comment:
even if we're confident that the genesis_config array will have a `0` index I would not access the index directly since it could panic. It's better to use something like `get()` and return an Err in case the index it's not available.